eradiate/scenes/measure/_core.py

```diff
@@ -466,20 +465,55 @@ def to_dataset(self, aggregate_spps: bool = False) -> xr.Dataset:
         if not aggregate_spps:
             return result
 
-        # Aggregate SPPs and drop sensor ID dimension
-        # Note: This will not work if multiple sensors are used for a purpose
-        # other than SPP splitting; should this happen, this part must be
-        # updated
-        weights = result.data_vars["spp"]
-        result_aggregated = result.assign(
-            {"raw": result["raw"].weighted(weights).mean(dim="sensor_id")},
+        measure_ids_sensor = sorted(
+            set(
+                [
+                    measure_id.split("_spp")[0]
+                    for measure_id in list(result["sensor_id"].values)
+                ]
+            )
         )
-        result_aggregated = result_aggregated.assign(
-            {"spp": result_aggregated.data_vars["spp"].sum(dim="sensor_id")},
+
+        raw_aggregated = []
+        spp_aggregated = []
+        for id_sensor in measure_ids_sensor:
+            # filter out the sensor ids that vary only in the spp part
+            relevant_ids = [
+                id for id in result.coords["sensor_id"].values if id_sensor in id
+            ]
+            # results are weighted by the respective SPPs
+            weights = xr.where(
+                result.coords["sensor_id"].isin(relevant_ids),
+                result.data_vars["spp"],
+                0,
+            )
+            raw_aggregated.append(
+                result["raw"]
+                .weighted(weights)
+                .mean(dim="sensor_id")
+                .expand_dims({"sensor_id": [id_sensor]}, axis=1)
+            )
+            # spps themselves must not be scaled, but just masked to include
+            # the right values in the sum
+            weights_spp = xr.where(result.coords["sensor_id"].isin(relevant_ids), 1, 0)
+            spp_aggregated.append(
+                result.data_vars["spp"]
+                .weighted(weights_spp)
+                .sum(dim="sensor_id")
+                .expand_dims({"sensor_id": [id_sensor]}, axis=1)
+            )
+
+        raw_redone = xr.combine_by_coords(raw_aggregated)
+        spp_redone = xr.combine_by_coords(spp_aggregated)
+
+        result_aggregated = xr.Dataset(
+            data_vars={
+                "raw": raw_redone["raw"],
+                "spp": spp_redone["spp"],
+                "sensor_id": list(measure_ids_sensor),
+            }
```

Comment:
```suggestion
        result_aggregated = xr.Dataset(
            data_vars={
                "raw": raw,
                "spp": spp,
                "sensor_id": sensor_id_prefixes,
            }
```
(`sorted()` already returns a list)